### DISCUSSION TITLE
[Feature] Allow for easier adding of non-detected / unknown faces in the GUI

### DISCUSSION THREAD

**Discussion by Gulpman** [immich-app/immich#14138] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi, everyone!

# Issue to solve
After a while, when you already have detected (a lot of) faces in Immich, it is a ton of scrolling down in the Explore view to add new, unknown to immich, faces. This is really time-consuming. It would be great to have a solution, which allows the adding of (to immich) unknown faces much quicker. This would make the migration to immich much faster and more convenient to the user.
## Ideas to solve this
One (or maybe more) of the following implementations would help in the process.
### Add settings for explorer view under preferences 
This could be done as shown in the following mockup:
![immich_explorer preferences](https://github.com/user-attachments/assets/a4423623-b3d0-4d6a-9b06-40ed132670b8)
This would be the most powerful way (I can imagine of ;) ) to support the adding of unknown faces. It would have the advantage that - dependent on what you want to achieve - you can choose how the explorer view will be used:

- usage phase -> sort by known faces first, as you may want to find those you are looking
- add new faces phase -> sort by unknown faces first, as you may want to add those unknown to the system.

### Add the possibility to filter for unknown faces directly in explorer view.
This could be done with an extra section for explorer view in preferences as shown in the following mockup: 
![immich_show undetected](https://github.com/user-attachments/assets/7e8ae6c8-13e1-448b-9e11-0c5b733bac2e)
I think this would be the least effort to implement - though it would have to be done every time again, if you want to filter for undetected faces. This would be a solution for thoses having not too many new faces to sort into Immich. 
### Change button "Show & hide people" in explore view to a drop-down menu with further possibilities
This could look like in the following mockup:
![immich_explorer dropdown](https://github.com/user-attachments/assets/46ba3ee4-685c-4bf2-b757-c5a6c821a419)
I am not a 100% sure if this position and a drop-down list is the best way of implementing this but as this screen is the one which always is reloaded after you add a new face, I think this could be a good place to implement a view for undetected faces.

## Conclusion
It would be great to have a feature which supports the process of adding new faces so one doesn't have to scroll through pages of already detected faces to get to the undetected ones.

I personally do prefer the first option of my ideas: In my opinion it is clean, easy to understand, doesn't change any default behavior and allows for easy switching betweeen "normal" mode and "adding unknown faces" mode.

Maybe there is another, much better way to solve this, that I can't think of right now. But it would be great to have better support for the process of adding unknown faces.

Best,
Sascha

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
I think this is (at least spiritually) a duplicate of #12328, right?

**Reply by Gulpman** [reply]
> I think this is (at least spiritually) a duplicate of #12328, right?

Hi @bo0tzz 
Indeed, I agree about the "spiritually" twin :) in #12328 after having a look at it.
I did a search before entering this but the author of my twin used different wordings so I didn't find it. 
I think the solution @Biepa proposed would work as well. Though I am not a 100% convinced, that I would prefer the solution proposed there. But her solution is less effort to implement, I think. 
So I'm fine to mark this as duplicate of #12328 but keep the solution to the problem mentioned open for discussion to solve this.

Best,
Sascha

**Comment by Gulpman** [comment]
Closed as duplicate of #12328
